agents/pysmurf_archiver/pysmurf_archiver_agent.py

```diff
@@ -50,11 +54,18 @@ def create_local_path(file, data_dir):
                 f"{str(dt.timestamp()):.5}",
                 f"{file['type']}"
              )
+    new_path = os.path.join(subdir, filename)
+    if os.path.exists(new_path):
+        print(f"Warning! Trying to archive duplicate of {new_path}. ")
+        subdir = os.path.join(subdir, 'duplicates')
+        new_path = os.path.join(subdir,
+                                f'{int(time.time())}_{filename}')
```

Comment:
You do not check if this exists.  And it's not guaranteed to be a new name.  One could imagine a failure pattern where a given file is registered multiple times in rapid succession (e.g. due to a save+reg being in a loop when it should be after the loop).  These entries would appear in the database together, and be processed together.  Consider starting with index=1, and incrementing index until f'{filename}.{index}' does not exist?